### PR TITLE
RISK=low RW-12216  Remove handleTerraShibbolethCallback

### PR DESCRIPTION
**Pull request by Qi77Qi**
Checked data access requirement page and no issue observed

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 1]
```diff
@@ -332,55 +330,6 @@ export const renewalRequiredModules: AccessModule[] = [
...
-    spinnerProps.showSpinner();
-    await profileApi().updateNihToken({ jwt: token });
-    spinnerProps.hideSpinner();
-    reloadProfile();
-  });
-
-  return handler();
-};
-
-const handleRasCallback = (
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:357]: we do not want to remove this

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 2]
```diff
@@ -603,8 +552,6 @@ export const DataAccessRequirements = fp.flow(withProfileErrorModal)(
     } = useStore(serverConfigStore);
 
     const urlParams = new URLSearchParams(window.location.search);
-    const code = urlParams.get('code');
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:606]: need this for RAS / id.me / login.gov

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 3]
```diff
@@ -698,27 +645,6 @@ export const DataAccessRequirements = fp.flow(withProfileErrorModal)(
...
-          variables.
-       */
-    // handle the route /nih-callback?token=<token>
-    useEffect(() => {
-      if (token) {
-        handleTerraShibbolethCallback(token, spinnerProps, reload);
-      }
-    }, [token]);
-
-    // handle the route /ras-callback?code=<code>
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:715]: restore this

**Review by jmthibault79 — CHANGES_REQUESTED** [review]
We are removing ERA/NIH/Shibboleth but not RAS/login.gov/id.me - please restore the latter.

We should also remove the code that directs users to the NIH Callback code removed here.  See `redirectToNiH` and `NIH_CALLBACK_PATH` in access-utils.tsx

File: ui/src/app/pages/access/data-access-requirements.tsx (hunk 1 above)
Comment (Qi77Qi) [reply ui/src/app/pages/access/data-access-requirements.tsx:357]: mm I see..somehow I thought era is for RAS linking or sth

Comment (jmthibault79) [reply ui/src/app/pages/access/data-access-requirements.tsx:357]: You *can* use RAS for ERA linking, but Shibboleth is more typical.  We also need RAS for the other identiy providers.

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 4]
```diff
@@ -698,20 +674,6 @@ export const DataAccessRequirements = fp.flow(withProfileErrorModal)(
         .finally(() => spinnerProps.hideSpinner());
     }, []);
 
-    /*
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:701]: this comment still applies - we should restore it

File: ui/src/app/pages/access/data-access-requirements.tsx (hunk 3 above)
Comment (Qi77Qi) [reply ui/src/app/pages/access/data-access-requirements.tsx:715]: updated!

**Pull request merged**